frame/support/src/storage/generator/map.rs

```diff
@@ -367,4 +267,130 @@ impl<K: FullEncode, V: FullCodec, G: StorageMap<K, V>> storage::StorageMap<K, V>
 			value
 		})
 	}
+
+	fn iter() -> storage::PrefixIterator<(K, V)> where
+		Self::Hasher: ReversibleStorageHasher
+	{
+		let prefix = G::prefix_hash();
+		storage::PrefixIterator {
+			prefix: prefix.clone(),
+			previous_key: prefix,
+			drain: false,
+			closure: from_hashed_key_to_key_value::<_, _, G>,
+		}
+	}
+
+	fn drain() -> storage::PrefixIterator<(K, V)> where
+		Self::Hasher: ReversibleStorageHasher
+	{
+		let prefix = G::prefix_hash();
+		storage::PrefixIterator {
+			prefix: prefix.clone(),
+			previous_key: prefix,
+			drain: true,
+			closure: from_hashed_key_to_key_value::<_, _, G>,
+		}
+	}
+
+	fn iter_values() -> storage::PrefixIterator<V> {
+		let prefix = Self::prefix_hash();
+		storage::PrefixIterator {
+			prefix: prefix.to_vec(),
+			previous_key: prefix.to_vec(),
+			drain: false,
+			closure: |_raw_key, mut raw_value| V::decode(&mut raw_value),
+		}
+	}
+
+	fn translate<O: Decode, F: Fn(K, O) -> Option<V>>(f: F) where
+		Self::Hasher: ReversibleStorageHasher
+	{
+		let prefix = G::prefix_hash();
+		let mut previous_key = prefix.clone();
+		loop {
+			match sp_io::storage::next_key(&previous_key).filter(|n| n.starts_with(&prefix)) {
+				Some(next) => {
+					previous_key = next;
+					let maybe_value = unhashed::get::<O>(&previous_key);
+					let value = match maybe_value {
+						Some(value) => value,
+						None => {
+							crate::debug::error!(
+								"next_key returned a key with no value at {:?}", previous_key
+							);
+							continue
+						},
+					};
+
+					// Slice in bound because already check by `starts_with`
+					let mut hashed_key = &previous_key[prefix.len()..];
+
+					let mut key_material = G::Hasher::reverse(&mut hashed_key);
+					let key = match K::decode(&mut key_material) {
+						Ok(key) => key,
+						Err(e) => {
+							crate::debug::error!(
+								"old key failed to decode at {:?}: {:?}",
+								previous_key, e
+							);
+							continue
+						},
+					};
+
+					match f(key, value) {
+						Some(new) => unhashed::put::<V>(&previous_key, &new),
+						None => unhashed::kill(&previous_key),
+					}
+				}
+				None => return,
+			}
+		}
+	}
+
+	fn translate_values<OldV, TranslateV>(translate_val: TranslateV) -> Result<(), u32>
+		where OldV: Decode, TranslateV: Fn(OldV) -> V
+	{
+		let prefix = Self::prefix_hash();
+		let mut previous_key = prefix.to_vec();
+		let mut errors = 0;
+		while let Some(next_key) = sp_io::storage::next_key(&previous_key)
+			.filter(|n| n.starts_with(&prefix[..]))
+		{
+			if let Some(value) = unhashed::get(&next_key) {
+				unhashed::put(&next_key[..], &translate_val(value));
+			} else {
+				// We failed to read the value. Remove the key and increment errors.
+				unhashed::kill(&next_key[..]);
+				errors += 1;
+			}
+
+			previous_key = next_key;
+		}
+
+		if errors == 0 {
+			Ok(())
+		} else {
+			Err(errors)
+		}
+	}
```

Comment:
this function has the old signature and implementation of PrefixedStorage::translate_values.
Change is planned in a following PR